pkg/command/context.go

```diff
@@ -212,33 +223,41 @@ func createNewContext() (context *configtypes.Context, err error) {
 		return createContextWithKubeconfig()
 	}
 	// user provided command line options to create a context using endpoint
-	if endpoint != "" || selfManaged {
-		return createContextWithEndpoint()
+	if contextType == "application-engine" || (endpoint != "" && isGlobalUCPEndpoint(endpoint)) {
```

Comment:
nit: suggesting a bit of refacting: how about defining the various creation type strings (""Mission Control", "Kubernetes Local Kubeconfig".. referenced in L250+) as constants, then also use them here to possibly assign to a local var initialized to "", call promptContextType only if we cannot detect what to use, and in all cases just return the result of createContextUsingContextType?

Reply:
Thanks! updated.